vaadin-checkbox-flow-parent/vaadin-checkbox-testbench/src/main/java/com/vaadin/flow/component/checkbox/testbench/CheckboxGroupElement.java

```diff
@@ -0,0 +1,140 @@
+import com.vaadin.testbench.TestBenchElement;
+import com.vaadin.testbench.elementsbase.Element;
+
+@Element("vaadin-checkbox-group")
+public class CheckboxGroupElement extends TestBenchElement
+        implements HasHelper {
+
+    /**
+     * Gets a list of all available options.
+     *
+     * @return a list of the options (visible text)
+     */
+    public List<String> getOptions() {
+        return getCheckboxes().stream().map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets all checkboxes which are part of this group.
+     *
+     * @return a list of all checkboxes in this group
+     */
+    public List<CheckboxElement> getCheckboxes() {
+        return $(CheckboxElement.class).all();
+    }
+
+    /**
+     * Selects the first checkbox matching the given text. If it is selected
+     * does nothing.
+     *
+     * @param text
+     *            the text of the option to select
+     */
+    public void selectByText(String text) {
+        Optional<CheckboxElement> checkbox = getCheckboxByText(text);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No item with text '" + text + "' found");
+        }
+
+        checkbox.get().setChecked(true);
+    }
+
+    /**
+     * Unselects the first checkbox matching the given text. If it is unslected
+     * does nothing.
+     *
+     * @param text
+     *            the text of the option to select
+     */
+    public void unselectByText(String text) {
+        Optional<CheckboxElement> checkbox = getCheckboxByText(text);
+        if (!checkbox.isPresent()) {
+            throw new NoSuchElementException(
+                    "No item with text '" + text + "' found");
+        }
+
+        checkbox.get().setChecked(false);
+    }
+
+    public List<String> getSelectedTexts() {
+        Stream<CheckboxElement> button = getSelectedCheckboxes();
+        return button.map(CheckboxElement::getLabel)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the selected checkboxes.
+     *
+     * @return an stream with the selected checkboxs elements.
+     */
+    private Stream<CheckboxElement> getSelectedCheckboxes() {
+        return getCheckboxByChecked(true);
+    }
+
+    /**
+     * Gets the checkbox which is part of this group and has the given visible
+     * text.
+     *
+     * @param text
+     *            the text to search for
+     * @return an optional with the checkbox element or an empty optional if no
+     *         match was found
+     */
+    private Optional<CheckboxElement> getCheckboxByText(String text) {
+        if (text == null) {
+            return Optional.empty();
+        }
+        return getCheckboxes().stream()
+                .filter(checkbox -> text.equals(checkbox.getLabel()))
+                .findFirst();
+    }
+
+    /**
+     * Gets the stream of checkboxes which is part of this group and has the
```

Comment:
nit: could mention `checked state` instead of value 🤔 

Should we consider something similar for indeterminate state or it would be an overkill?

Reply:
> could mention checked state instead of value 

Done